Network/DNS/Resolver.hs

```diff
@@ -173,13 +173,17 @@ makeResolvSeed conf = do
 
 getDefaultDnsServers :: FilePath -> IO [String]
 #if defined(WIN)
+foreign import ccall "getWindowsDefDnsServers" getWindowsDefDnsServers :: CString -> Int -> IO Word32
 getDefaultDnsServers _ = do
-  res <- peek =<< getWindowsDefDnsServers
-  case dnsError res of
-    0 -> return $ map T.unpack (T.splitOn "," (T.pack (dnsAddresses res)))
-    _ -> do
-      -- TODO: Do proper error handling here.
-      return mempty
+  allocaBytes 128 $ \cString -> do
+     res <- getWindowsDefDnsServers cString 128
+     case res of
+       0 -> do
+         addresses <- BS.pack <$> peekCString cString
```

Comment:
Is this a conversion of `CString` => `String` => `ByteString`?
If so, can't we use `Data.ByteString.packCString`?

Reply:
I think that you should use `String` directly instead of `ByteString`.
Processing `String` might be slow but this is called only once and the length of `String` is small.
So, we don't have to worry about performance here.

Reply:
`Data.List.Split.split` in `split` would be the function which you are looking for.

```haskell
         addresses <- peekCString cString
         return $ filter (not . null) $ split ',' address
```

Reply:
@kazu-yamamoto To be honest, I cannot recall why I have used a `ByteString` in the first place! It might be that I wanted to tap into a richer API than `Data.List`, but since @dcoutts rightly pointed out I can use stuff from `Foreign.Marshal.Alloc` I don't think we need it.

> Data.List.Split.split in splitOn would be the function which you are looking for.

But this would introduce a dependency on `split` wouldn't it? I guess if you are happy with that, I'm happy 😉